mozilla/blurts-server
# MNTOR-1692 Resolution Flow: Data Broker Results

## Summary
This PR adds initial routes for the entire guided fix flow, and builds out the initial Data Broker scan results page. Additional pages (welcome, manual removal, automatic removal) 

- Jira: [MNTOR-1692](https://mozilla-hub.atlassian.net/browse/MNTOR-1692)

### TODO
- [ ] ~Add logic to next/prev buttons~ _See known issues_
- [ ] ~Add logic/state to ℹ️ button next to _"View your info on these sites"_ text~ _See known issues_
- [x] Add logic to "view more" button to show additional profiles
- [x] Add close/X button 
- [x] Style "Remove them for me" page (No functionality) 
- [x] Style "I'll remove them manually" page (No functionality) 
- [x] Add logic/active state to progress bar 
- [x] ~Add (Number) on progress bar~ _See known issues_

Known issues: 
- Setting the correct target forward/backward arrow links is not working. Root issue is component placement to pass targets on a per page/route level.  This same sort of issue also applies to the (#) in the progress bar. 
- Cannot add the [standard dialog modal](https://github.com/mozilla/blurts-server/tree/main/src/app/components/client/dialog) to my page as the `page.tsx` are currently server-side
- Logos are missing for data brokers 
- The logic isn't very protected. You can "land" in this flow by going to the URL, regardless if you've started a scan

#### Other Notes
- When showing data counts for each data broker profile, be sure the number is generated/served from the server to not expose extra info 

## Testing 

⭐ Setup: You need scan results first: Please run through the welcome onboarding flow (`/redesign/user/welcome`) first. 

Testing steps: 
- Navigate to `/redesign/user/dashboard/`
  - Alternatively, you could start on `redesign/user/dashboard/fix/data-broker-profiles/start-free-scan`
- Click on "Let's fix it" in the top banner on the dashboard 
  - Note: This should take you to `/redesign/user/dashboard/fix/data-broker-profiles/view-data-brokers`
- Expected: You should get the [data broker screen](https://www.figma.com/file/CaEKIhvSJqf6KNIMzSkt40/Concepts-for-Monitor-MVP-Redesign?type=design&node-id=1290-58160&mode=dev)
- From here, you can 
  - View all the data broker profiles (Click "view more") 
  - Go to upgrade flow (Click "Remove them for me") 
  - Go to manual removal flow (Click "I'll remove them manually") 

## Screenshots

TBD
